### PR TITLE
Extended the coroutines feature documentation

### PR DESCRIPTION
**Pull request by thedrow**

### CODE REVIEW

File: docs/definition.md [hunk 1]
```diff
@@ -180,6 +180,573 @@ There are a few terms you should be familiar with:
...
+    ```pycon tab="sync"
+
+    >>> from stories import Failure, Result, Success, arguments, story
+    
+    >>> class Subscription:
+    ...     """Buy subscription for certain category."""
+    ...
+    ...     @story
+    ...     @arguments("category_id", "price_id", "user_id")
+    ...     async def buy(I):
```
Comment (proofit404) [inline docs/definition.md]: Maybe this example is way too long to show the problem. Can we define it with fever stories steps?

File: docs/definition.md [hunk 2]
```diff
@@ -180,6 +180,573 @@ There are a few terms you should be familiar with:
 7. `Success`, `Failure` and `Result` are markers returned by step
    methods to change business process execution path.
 
+### Invalid story definitions
```
Comment (proofit404) [inline docs/definition.md]: Does it make more sense to move it to the FAQ section?

Comment (thedrow) [reply docs/definition.md]: No, this should be present here so our users will know what not to do.
This is as important as what you can do.

File: docs/definition.md (hunk 1 above)
Comment (thedrow) [reply docs/definition.md]: Sure, I was simply following along :)

File: tests/helpers/mddoctest.py [hunk 3]
```diff
@@ -38,7 +39,9 @@ def _main():
     markdown_files = glob("**/*.md", recursive=True)
     exit_code = 0
     for markdown_file in markdown_files:
-        failed, attempted = testfile(markdown_file, module_relative=False)
+        failed, attempted = testfile(
+            markdown_file, module_relative=False, optionflags=IGNORE_EXCEPTION_DETAIL
```
Comment (proofit404) [inline tests/helpers/mddoctest.py]: Why is this necessary?

Comment (proofit404) [reply tests/helpers/mddoctest.py]: I corrected doctest examples so this is no longer necessary.

**Comment by proofit404** [comment]
Thanks for the PR :tada:
